sphinx/application.py

```diff
@@ -143,8 +145,8 @@ def __init__(self, srcdir: str | os.PathLike[str], confdir: str | os.PathLike[st
                  status: IO[str] | None = sys.stdout, warning: IO[str] | None = sys.stderr,
                  freshenv: bool = False, warningiserror: bool = False,
                  tags: Sequence[str] = (),
-                 verbosity: int = 0, parallel: int = 0, keep_going: bool = False,
-                 pdb: bool = False) -> None:
+                 verbosity: int = 0, parallel: int = 0,
```

Comment:
Could you keep the `keep_going` argument? it's quite important because it's a *positional* argument. And because of that, it's important for plugins that *could* have relied on the position.

The PR I have for the pytest plugin tries to be as compatible as possible with bad constructions and with how the application is being configured. I'd rather appreciate not to do anything in this PR concerning the argument's order.

And we should rather check which combination of arguments is legit or not in the constructor itself.